CMakeLists.txt

```diff
@@ -0,0 +1,129 @@
+cmake_minimum_required(VERSION 3.18)
+project(nx_hbmenu VERSION 3.6.0)
+
+if (NOT NINTENDO_SWITCH)
+  find_package(SDL2 REQUIRED)
+  find_package(SDL2 CONFIG COMPONENTS SDL2main)
+endif()
+
+find_package(Freetype REQUIRED)
+find_package(PNG REQUIRED)
+find_package(PhysFS REQUIRED)
+find_package(libjpeg-turbo REQUIRED)
+find_package(libconfig REQUIRED)
+
+add_compile_options(
+    -DVERSION=\"${PROJECT_VERSION}\"
+)
+
+add_executable(${PROJECT_NAME})
+
+add_custom_command(
+  OUTPUT ${CMAKE_SOURCE_DIR}/romfs/assets.zip
+  COMMAND mkdir -p ${CMAKE_SOURCE_DIR}/romfs
```

Comment:
Might be an idea to use cmake in tool mode for compatibility.
`${CMAKE_COMMAND} -E`, it has rm, make_directory and tar (which can even work on zip files).